internal/engine/analytics_reporter.go

```diff
@@ -13,14 +15,25 @@ import (
 // How often to periodically report data for analytics while Tilt is running
 const analyticsReportingInterval = time.Hour * 1
 
+const newAnalyticsFlag = "TILT_NEW_ANALYTICS"
+
+// TEMPORARY: check env to see if new-analytics flag is set
+func newAnalyticsOn() bool {
+	return os.Getenv(newAnalyticsFlag) != ""
+}
+
 type AnalyticsReporter struct {
 	a       analytics.Analytics
 	store   *store.Store
 	started bool
+	opt     analytics.Opt
 }
 
 func (ar *AnalyticsReporter) OnChange(ctx context.Context, st store.RStore) {
 	if ar.started {
+		if newAnalyticsOn() {
```

Comment:
This might be cleaner as a separate class? Nudging is a pretty separable responsibility from periodically reporting.

This feels a bit weird to me. Previously, `AnalyticsReporter` was unaware of optin status. Now it's aware, but sends data to `ar.a` even if the user has opted out (because `ar.a` itself is aware of the opt-in status and discards data if the user has opted out).

I don't know if this would be considered dirty, but this logic might even be simple enough to just be in a new function called from `handleBuildCompleted`?

Reply:
sigh. i hate that you may be right. but you may be right.